### PR TITLE
Added hash function description

### PR DESCRIPTION
**Pull request by hwwhww**
* Added a section for hash function description.
* Not sure where is good to put it... so I add Appendix section currently, it can be changed.

### CODE REVIEW

File: specs/casper_sharding_v2.1.md [hunk 1]
```diff
@@ -441,5 +441,14 @@ Slashing conditions may include:
     Proof of custody no secret reveal
     RANDAO leak
 
+# Appendix
+## Appendix A - Hash function
+The general hash function `hash(x)` in this specification is defined as: 
+
+`hash(x) := BLAKE2b(x)[0:32]`, where `BLAKE2b` algorithm is deifned in [RFC7693](https://tools.ietf.org/html/rfc7693) and input `x` is bytes type.
```
Comment (mratsim) [inline specs/casper_sharding_v2.1.md]: defined

Comment (hwwhww) [reply specs/casper_sharding_v2.1.md]: Thanks for the catch!

**Comment by hwwhww** [comment]
@terenc3t @mratsim thanks for the reviews! I updated a bit. 🙂

**Comment by djrtwo** [comment]
lgtm!

**Pull request merged**